src/Glide64/3dmath.cpp

```diff
@@ -190,16 +190,37 @@ void InverseTransformVectorC (float *src, float *dst, float mat[4][4])
 
 void MulMatricesC(float m1[4][4],float m2[4][4],float r[4][4])
 {
-  for (int i=0; i<4; i++)
-  {
-    for (int j=0; j<4; j++)
+    float row[4][4];
+    register unsigned int i, j;
+
+    for (i = 0; i < 4; i++)
+        for (j = 0; j < 4; j++)
+            row[i][j] = m2[i][j];
+    for (i = 0; i < 4; i++)
     {
-      r[i][j] = m1[i][0] * m2[0][j] +
-                m1[i][1] * m2[1][j] +
-                m1[i][2] * m2[2][j] +
-                m1[i][3] * m2[3][j];
+        float leftrow[4], destrow[4];
```

Comment:
That's fine.  In particular, my point is, only the C code (also thought of as the "auto-vectorized" code) has universal meaning.  Anything else like inline assembly, or assembly language intrinsics, is in the eye of the beholder and not naturally readable code to everyone--because, its readability is dictated more by the target processor.

For example, suppose you might prefer the intrinsic use:
`ALIGNED int32_t a[4], b[4], c[4];
int main(void) {
    *(__m128i *)c = _mm_and_si128(*(__m128i *)a, *(__m128i *)b);
    return 0;
}`

However, I find it even more readable to both the human and the compiler (for better optimizations outside the version of the SIMD instruction set architecture assumed by the usage of that intrinsic) to instead say:

`int a[4], b[4], c[4];
int main(void) {
    register unsigned int i;

    for (i = 0; i < 4; i++)
        c[i] = a[i] & b[i];
    return 0;
}`

Although the name of the `_mm_and_si128` intrinsic is self-explanatory as to the operation of the code, the algorithm itself is more readable and more natural when using C code (and the compiler is left to generate what should be the best choice of SIMD intrinsic for and'ing all elements).  This commit probably isn't the best example of that, however--you're welcome to disagree whether my C loops are as readable as some of the old code.  The gist of it may have been portability, but I thought it was more readable to divide each parallelized step of the overal algorithm into its own loop.  The original C loop code you had was much better than mine, thougin terms of shortness and brevity--maybe open to interpretation on the parallel steps what goes on.  So I think it's fine if certain sketchy examples of this practice come up that you like to comment about.